### PR TITLE
Fix sass issues, default sourceRoot, adjust ts sourceMaps and prevent…

### PR DESCRIPTION
**Pull request by zigmo**
… vendor sourceMaps if not enabled. Fix for #778

### CODE REVIEW

File: src/plugins/stylesheet/SassPlugin.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export class SassPluginClass implements Plugin {
 
         const options = Object.assign({
             data: file.contents,
+            file: context.homeDir+"/"+file.info.fuseBoxPath,
```
Comment (nchanged) [inline src/plugins/stylesheet/SassPlugin.ts]: In this case we need to add it to Post and Less too! It's not just the only plugin.

File: src/core/File.ts [hunk 2]
```diff
@@ -590,11 +592,7 @@ export class File {
             let jsonSourceMaps = JSON.parse(this.sourceMap);
             jsonSourceMaps.file = this.info.fuseBoxPath;
             jsonSourceMaps.sources = jsonSourceMaps.sources.map((source : string) => {
-                if(source.indexOf('stdin') !== -1) {
```
Comment (nchanged) [inline src/core/File.ts]: Anything we can do about stdin? Why did you remove it?

File: src/BundleSource.ts [hunk 3]
```diff
@@ -139,6 +149,11 @@ export class BundleSource {
             return;
         }
 
+        if(!this.includeSourceMaps) {
+            // set to true if this file has relevant sourceMaps
+            this.includeSourceMaps = file.belongsToProject() && this.context.sourceMapsProject || !file.belongsToProject() && this.context.sourceMapsVendor;
```
Comment (nchanged) [inline src/BundleSource.ts]: 👍

File: src/plugins/stylesheet/SassPlugin.ts (hunk 1 above)
Comment (zigmo) [reply src/plugins/stylesheet/SassPlugin.ts]: This need to be checked. Don't know if its required. This is a [special behaviour](https://github.com/sass/node-sass#special-behaviours) of node-sass. `file` is only used for sourceMaps if data is provided.

File: src/core/File.ts (hunk 2 above)
Comment (zigmo) [reply src/core/File.ts]: `stdin` shouldn't be there anymore because `file` is additionally provided to node-sass options - we will have the right file name with relativ path instead `stdin`. See SassPlugin comment.

**Comment by zigmo** [comment]
Cool - maybe this can be a separated/expiremental pre-beta-release or so - so we can better test it. I think this would also solve this issue #543

**Comment by nchanged** [comment]
Ok. it's here `fuse-box@2.3.1-test.1` @rs3d

**Comment by rs3d** [comment]
Great job and thank you for the help @zigmo @nchanged!  The file structure for source files is working now.

**Some minor issues:**
1. `file":"index.scss"`should be named `index.css` to properly fit the generated filename, right?https://github.com/rs3d/fuse-box-sass-issue/blob/master/dist/index.scss.map#L1 
image



2.  Maybe a different issue: `outFile` without using `file` as filename: 
```
CSSPlugin({
         outFile: file => `${distPath}/module-XXX.css`
}) 
```
This generates `module-XXX.css` and in the same folder `module-XXX.css.map`.
But the resulting source map path is: 
`/*# sourceMappingURL=index.scss.map */` which should be 
`/*# sourceMappingURL=module-XXX.css.map */` to make this fully working.

**Comment by nchanged** [comment]
@rs3d 
`outFile: file => `${distPath}/module-XXX.css`
that's dynamic

You should use it like so:

```
outFile: file => `${distPath}/${ file}.css
```

**Comment by rs3d** [comment]
@nchanged Yes I know this isn't how it's intended. In my case it's a whitelabel platform in different languages, so my css files have a platform specific prefix. 

Maybe this is a more legit example?
```
outFile: file => `${distPath}/platform-${file}
```
Or for different languages (like arabic):
```
outFile: file => `${distPath}/ar-AE-${file}
```

**Comment by rs3d** [comment]
Never mind, I'm switching to folder based structure now :) So i  don't need that many workarounds. Thx.

**Comment by nchanged** [comment]
Cool!
